libsolidity/interface/CompilerStack.cpp

```diff
@@ -480,154 +485,185 @@ bool CompilerStack::analyze()
-					noErrors = false;
-		}
+	if (noErrors)
+	{
+		// Run SMTChecker
 
-		if (noErrors)
-		{
-			// Check for state mutability in every function.
-			std::vector<ASTPointer<ASTNode>> ast;
-			for (Source const* source: m_sourceOrder)
-				if (source->ast)
-					ast.push_back(source->ast);
+		auto allSources = util::applyMap(m_sourceOrder, [](Source const* _source) { return _source->ast; });
+		if (ModelChecker::isPragmaPresent(allSources))
+			m_modelCheckerSettings.engine = ModelCheckerEngine::All();
 
-			if (!ViewPureChecker(ast, m_errorReporter).check())
-				noErrors = false;
-		}
+		// m_modelCheckerSettings is spread to engines and solver interfaces,
+		// so we need to check whether the enabled ones are available before building the classes.
+		if (m_modelCheckerSettings.engine.any())
+			m_modelCheckerSettings.solvers = ModelChecker::checkRequestedSolvers(m_modelCheckerSettings.solvers, m_errorReporter);
 
-		if (noErrors)
-		{
-			// Run SMTChecker
-
-			auto allSources = util::applyMap(m_sourceOrder, [](Source const* _source) { return _source->ast; });
-			if (ModelChecker::isPragmaPresent(allSources))
-				m_modelCheckerSettings.engine = ModelCheckerEngine::All();
-
-			// m_modelCheckerSettings is spread to engines and solver interfaces,
-			// so we need to check whether the enabled ones are available before building the classes.
-			if (m_modelCheckerSettings.engine.any())
-				m_modelCheckerSettings.solvers = ModelChecker::checkRequestedSolvers(m_modelCheckerSettings.solvers, m_errorReporter);
-
-			ModelChecker modelChecker(m_errorReporter, *this, m_smtlib2Responses, m_modelCheckerSettings, m_readFile);
-			modelChecker.checkRequestedSourcesAndContracts(allSources);
-			for (Source const* source: m_sourceOrder)
-				if (source->ast)
-					modelChecker.analyze(*source->ast);
-			m_unhandledSMTLib2Queries += modelChecker.unhandledQueries();
-		}
-	}
-	catch (FatalError const&)
-	{
-		if (m_errorReporter.errors().empty())
-			throw; // Something is weird here, rather throw again.
-		noErrors = false;
+		ModelChecker modelChecker(m_errorReporter, *this, m_smtlib2Responses, m_modelCheckerSettings, m_readFile);
+		modelChecker.checkRequestedSourcesAndContracts(allSources);
+		for (Source const* source: m_sourceOrder)
+			if (source->ast)
+				modelChecker.analyze(*source->ast);
+		m_unhandledSMTLib2Queries += modelChecker.unhandledQueries();
 	}
 
-	if (!noErrors)
-		return false;
+	return noErrors;
+}
 
-	m_stackState = AnalysisSuccessful;
-	return true;
+bool CompilerStack::analyzeExperimental()
+{
+	bool noErrors = true;
+	solAssert(!m_experimentalAnalysis);
+	m_experimentalAnalysis = std::make_unique<experimental::Analysis>(m_errorReporter);
+	std::vector<std::shared_ptr<SourceUnit const>> sourceAsts;
+	for (Source const* source: m_sourceOrder)
+		if (source->ast)
+			sourceAsts.emplace_back(source->ast);
+	if (!m_experimentalAnalysis->check(sourceAsts))
+		noErrors = false;
+	return noErrors;
```

Comment:
We can actually just simplify this as
```suggestion
	return m_experimentalAnalysis->check(sourceAsts);
```
and just ditch the `noErrors` variable - since experimental Analysis is a class in itself, we don't have to expect that anything more will ever happen on this end in the compiler stack - not sure why I added the variable in the first place.